src/genesys/agentApp/public/js/WatsonAssistantAgentAppWidgetProxy.js

```diff
@@ -0,0 +1,192 @@
+//       So we have to check if we were able to parse out the environment or not.
+let integrationQueryString = '';
+if (window.location.search.length !== 0) {
+  integrationQueryString = window.location.search.substring(1);
+} else if (window.location.hash.length !== 0) {
+  integrationQueryString = window.location.hash.substring(1);
+}
+const appParams = parseAppParameters(integrationQueryString);
+
+console.log(`Initializing platform client for region: ${appParams.pcEnvironment}`);
+client.setEnvironment(appParams.pcEnvironment);
+
+// Create instance of Client App SDK
+const myClientApp = new window.purecloud.apps.ClientApp({
+  pcEnvironment: appParams.pcEnvironment,
+});
+
+// Log the PureCloud environment (i.e. AWS Region)
+console.log(`PureCloud API Client Environment: ${client.environment}`);
+console.log(`PureCloud ClientApp Environment: ${myClientApp.pcEnvironment}`);
+console.log(`PureCloud ClientApp Version: ${window.purecloud.apps.ClientApp.version}`);
+console.log(`PureCloud ClientApp About: ${window.purecloud.apps.ClientApp.about()}`);
+
+initializeApplication();
+
+//
+// Bootstrap Listener
+//
+myClientApp.lifecycle.addBootstrapListener(() => {
+  logLifecycleEvent('App Lifecycle Event: bootstrap', true);
+  initializeApplication();
+});
+
+//
+// Focus Listener
+//
+function onAppFocus() {
+  logLifecycleEvent('App Lifecycle Event: focus', true);
+
+  myClientApp.alerting.showToastPopup(lifecycleStatusMessageTitle, 'App Focused', {
+    id: lifecycleStatusMessageId,
+  });
+}
+myClientApp.lifecycle.addFocusListener(onAppFocus);
+
+//
+// Blur Listener
+//
+function onAppBlur() {
+  logLifecycleEvent('App Lifecycle Event: blur', true);
+
+  myClientApp.alerting.showToastPopup(lifecycleStatusMessageTitle, 'App Blurred', {
+    id: lifecycleStatusMessageId,
+  });
+}
+myClientApp.lifecycle.addBlurListener(onAppBlur);
+
+//
+// Stop Listener
+//
+myClientApp.lifecycle.addStopListener(() => {
+  logLifecycleEvent('App Lifecycle Event: stop', true);
+
+  // Clean up other, persistent listeners
+  myClientApp.lifecycle.removeFocusListener(onAppFocus);
+  myClientApp.lifecycle.removeBlurListener(onAppBlur);
+
+  myClientApp.lifecycle.stopped();
+
+  myClientApp.alerting.showToastPopup(lifecycleStatusMessageTitle, 'App Stopped', {
+    id: lifecycleStatusMessageId,
+    type: 'error',
+    showCloseButton: true,
+  });
+
+  logLifecycleEvent('Notified Genesys Cloud of Successful App Stop', false);
+});
+
+function logLifecycleEvent(logText, incommingEvent) {
+  console.log(logText);
+}
+
+function initializeApplication() {
+  console.log('Performing application bootstrapping');
+
+  // Perform Implicit Grant Authentication
+  //
+  // Note: Pass the query string parameters in the 'state' parameter so that they are returned
+  //       to us after the implicit grant redirect.
+  client
+    .loginImplicitGrant(GENESYS_CLIENT_ID, redirectUri, { state: integrationQueryString })
+    .then((data) => {
```

Comment:
If you are just copying code from their example I'm not sure if I feel strongly enough to suggest you change this, but it is preferable to use `async/wait` instead of `Promise.then/catch`.

Reply:
it was sample code, but I'll clean it up shouldn't be too much effort